rp2040-hal/src/multicore.rs

```diff
@@ -144,15 +144,21 @@ impl<'p> Core<'p> {
     }
 
     /// Spawn a function on this core.
+    ///
+    /// Upon exit of the entry function, the core will enter an infinite `wfe` (wait for event)
+    /// loop. In order to reduce power consumption, it is recommended that all interrupts and events
+    /// are disabled before returning.
```

Comment:
Alternative idea: Just say that the function shouldn't return, and it's only defined differently because the never type isn't stable yet.
(But documenting the infinite loop is fine as well.)